pkg/proxy/iptables/proxier.go

```diff
@@ -612,34 +612,8 @@ func (proxier *Proxier) OnEndpointSlicesSynced() {
 	proxier.syncProxyRules()
 }
 
-// OnNodeAdd is called whenever creation of new node object
-// is observed.
-func (proxier *Proxier) OnNodeAdd(node *v1.Node) {
-	if node.Name != proxier.hostname {
-		proxier.logger.Error(nil, "Received a watch event for a node that doesn't match the current node",
-			"eventNode", node.Name, "currentNode", proxier.hostname)
-		return
-	}
-
-	if reflect.DeepEqual(proxier.nodeLabels, node.Labels) {
-		return
-	}
-
-	proxier.mu.Lock()
-	proxier.nodeLabels = map[string]string{}
-	for k, v := range node.Labels {
-		proxier.nodeLabels[k] = v
-	}
-	proxier.needFullSync = true
-	proxier.mu.Unlock()
-	proxier.logger.V(4).Info("Updated proxier node labels", "labels", node.Labels)
-
-	proxier.Sync()
-}
-
-// OnNodeUpdate is called whenever modification of an existing
-// node object is observed.
-func (proxier *Proxier) OnNodeUpdate(oldNode, node *v1.Node) {
+// OnNodeUpsert is called whenever creation of new or modification of an existing node object is observed.
+func (proxier *Proxier) OnNodeUpsert(node *v1.Node) {
```

Comment:
Eventually, the Proxier-level signal should just be `OnTopologyChange(topologyLabels map[string]string)`, the same way that for ServiceCIDR we didn't expose Add/Update/Delete events, we just have a single `OnServiceCIDRsChanged(cidrs []string)`.

(One way to do that would be to have `config.NodeConfig` for `NodeManager` and a separate `config.NodeTopologyConfig` interface for `proxy.Provider`. Another way would be to not have the `proxy.Provider` watch nodes directly, and instead have it just receive updates from the `NodeManager`, somehow?)